modules/greenbidsAnalyticsAdapter.js

```diff
@@ -125,83 +145,82 @@ export const greenbidsAnalyticsAdapter = Object.assign(adapter({ANALYTICS_SERVER
     };
   },
   /**
-   * @param {Bid} bid
-   * @param {BIDDER_STATUS} status
-   */
-  serializeBidResponse(bid, status) {
-    return {
-      bidder: bid.bidder,
-      isTimeout: (status === BIDDER_STATUS.TIMEOUT),
-      hasBid: (status === BIDDER_STATUS.BID),
-      params: (bid.params && Object.keys(bid.params).length > 0) ? bid.params : {},
-      ...(status === BIDDER_STATUS.BID ? {
-        cpm: bid.cpm,
-        currency: bid.currency
-      } : {}),
-    };
-  },
-  /**
-   * @param {*} message Greenbids API payload
-   * @param {Bid} bid Bid to add to the payload
-   * @param {BIDDER_STATUS} status Bidding status
+   * @param {AuctionEndArgs} auctionEndArgs
+   * @param {GreenbidsCachedOption} cachedAuction
+   * @returns {Message}
    */
-  addBidResponseToMessage(message, bid, status) {
-    const adUnitCode = bid.adUnitCode.toLowerCase();
-    const adUnitIndex = message.adUnits.findIndex((adUnit) => {
-      return adUnit.code === adUnitCode;
-    });
-    if (adUnitIndex === -1) {
-      logError('Trying to add to non registered adunit');
-      return;
-    }
-    const bidderIndex = message.adUnits[adUnitIndex].bidders.findIndex((bidder) => {
-      return bidder.bidder === bid.bidder;
-    });
-    if (bidderIndex === -1) {
-      message.adUnits[adUnitIndex].bidders.push(this.serializeBidResponse(bid, status));
-    } else {
-      message.adUnits[adUnitIndex].bidders[bidderIndex].params = (bid.params && Object.keys(bid.params).length > 0) ? bid.params : {};
-      if (status === BIDDER_STATUS.BID) {
-        message.adUnits[adUnitIndex].bidders[bidderIndex].hasBid = true;
-        message.adUnits[adUnitIndex].bidders[bidderIndex].cpm = bid.cpm;
-        message.adUnits[adUnitIndex].bidders[bidderIndex].currency = bid.currency;
-      } else if (status === BIDDER_STATUS.TIMEOUT) {
-        message.adUnits[adUnitIndex].bidders[bidderIndex].isTimeout = true;
-      }
-    }
-  },
-  createBidMessage(auctionEndArgs) {
-    const {auctionId, timestamp, auctionEnd, adUnits, bidsReceived, noBids} = auctionEndArgs;
-    const cachedAuction = this.getCachedAuction(auctionId);
-    const message = this.createCommonMessage(auctionId);
-    const timeoutBids = cachedAuction.timeoutBids || [];
+  createBidMessage(auctionEndArgs, cachedAuction) {
+    const {
+      auctionId,
+      timestamp,
+      auctionEnd,
+      adUnits,
+      bidsReceived,
+      noBids
+    } = auctionEndArgs;
+    const timeoutBids = (cachedAuction || this.getCachedAuction(auctionId)).timeoutBids || [];
```

Comment:
question: the auctionId is cached twice here. Could we cache it once at `handleAuctionInit` and then reuse?